src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Extract.cs

```diff
@@ -103,6 +103,22 @@ public void Windows_ZipWithInvalidFileNames_ThrowsArgumentException(string zipNa
             AssertExtensions.Throws<ArgumentException>("path", null, () => ZipFile.ExtractToDirectory(compat(zipName) + ".zip", GetTestFilePath()));
         }
 
+        /// <summary>
+        /// This test checks whether or not ZipFile.ExtractToDirectory() is capable of handling filenames
+		/// which contain invalid path characters in Windows.
+        /// </summary>
+        [Theory]
+        [InlineData("InvalidWindowsFileNameChars.zip", "Test______________________________________.txt")]
+        [InlineData("InvalidWindowsFileNameChars.zip", "Test______________________________________/TestText1______________________________________.txt")]
+        [InlineData("InvalidWindowsFileNameChars.zip", "Test/normalText.txt")]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void Windows_ZipWithInvalidFileNames(string zipFile, string testPath)
```

Comment:
it would be helpful to include a comment here showing what the file paths are inside the zip. maybe just what I pasted into the runtime-assets PR you did.